storagenode/nodestats/cache.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package nodestats
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"storj.io/storj/internal/date"
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/storagenode/reputation"
+	"storj.io/storj/storagenode/storageusage"
+	"storj.io/storj/storagenode/trust"
+)
+
+var (
+	// NodeStatsCacheErr defines node stats cache loop error
+	NodeStatsCacheErr = errs.Class("node stats cache error")
+)
+
+// Cache runs cache loop and stores reputation stats
+// and storage usage into db
+type Cache struct {
+	log *zap.Logger
+
+	service        *Service
+	trust          *trust.Pool
+	reputationDB   reputation.DB
+	storageusageDB storageusage.DB
+
+	reputationCycle *sync2.Cycle
+	storageCycle    *sync2.Cycle
+}
+
+// NewCache creates new caching service instance
+func NewCache(log *zap.Logger, service *Service, trust *trust.Pool, reputationDB reputation.DB, storageusageDB storageusage.DB) *Cache {
+	return &Cache{
+		log:             log,
+		service:         service,
+		trust:           trust,
+		reputationDB:    reputationDB,
+		storageusageDB:  storageusageDB,
+		reputationCycle: sync2.NewCycle(time.Hour * 4),
```

Comment:
these times should be configurable. Add a struct like
```
type Config struct {
	ReputationSync time.Duration `help:"how often to sync reputation" releaseDefault:"4h" devDefault:"5m"`
	StorageSync time.Duration `help:"how often to sync storage" releaseDefault:"12h" devDefault:"5m"`
}
```

Then, take `config Config` as an argument to `NewCache`, and add it to the `Config` struct in `peer.go`.

Reply:
Yeah, it feels like it should be configurable, but the main reason this caching service exits is to reduce satellite load, so if intervals are configurable it is more likely that some `storagenodes` will query `satellite` to often

Reply:
People can modify the parameters anyways whether we put the flags in place or not. We should instead have rate-limiting on the satellite.